File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/users/dtos/SearchCriteria.java
package io.zenwave360.example.clinicaltool.modules.users.dtos;

import io.zenwave360.example.clinicaltool.modules.users.domain.*;
import java.io.Serializable;
import java.math.*;
import java.time.*;
import java.util.*;
import jakarta.validation.constraints.*;

/**
 * SearchCriteria.
 */
@lombok.Getter @lombok.Setter
public  class SearchCriteria  implements Serializable {


    
    private String username ;

    
    private String email ;

    
    private String role ;

    
    private Boolean enabled ;

    
    private Map<String, Object> searchTerms ;








}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/users/dtos/SearchCriteria.kt
package io.zenwave360.example.clinicaltool.modules.users.dtos

import io.zenwave360.example.clinicaltool.modules.users.domain.*
import jakarta.validation.constraints.*
import java.io.Serializable
import java.math.*
import java.time.*

/** SearchCriteria. */
data class SearchCriteria(
    val username: String? = null,
    val email: String? = null,
    val role: String? = null,
    val enabled: Boolean? = null,
    val searchTerms: MutableMap<String, Object>? = null,
) : Serializable {}
